File: Solutions/Cpp/remove_element.cpp
/*
    [EASY]
    27. Remove Element

    Concepts:
    - array

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 11.57 MB  [Beats 97.68%]
*/

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        for (auto it = nums.begin(); it != nums.end();) {
            if (*it == val) nums.erase(it);
            else it++;
        }
        return nums.size();
    }
};

File: Solutions/Rust/remove_element.rs
/*
    [EASY]
    27. Remove Element

    Concepts:
    - array

    Stats:
        Runtime | 0 ms      [Beast 100%]
        Memory  | 2.17 MB   [Beats 64.71%]
*/

impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        nums.retain(|&num| num != val);
        return nums.len() as i32;
    }
}